File: Lecture03/ActivityLifecycle/app/src/main/java/com/codingblocks/activitylifecycle/MainActivity.java
package com.codingblocks.activitylifecycle;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    public final String TAG = getClass().getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.e(TAG, "onCreate: ");
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.e(TAG, "onStart: ");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.e(TAG, "onResume: ");
    }

    //User is using the app

    @Override
    protected void onPause() {
        super.onPause();
        Log.e(TAG, "onPause: ");
        //Called when the app is partially visible for the first time
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.e(TAG, "onStop: ");
        //Called when the app is no longer visible on the screen
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.e(TAG, "onDestroy: ");
        //Called when the system decides to kill your app to free up some memory
    }


    @Override
    public void onBackPressed() {
        super.onBackPressed();
    }
}

File: Lecture12/SaveInstanceState/app/src/main/java/com/codingblocks/saveinstancestate/MainActivity.kt
package com.codingblocks.saveinstancestate

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.PersistableBundle
import android.util.Log
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

//    companion object {
//        var count = 0
//
//        fun returnCount() = count
//    }

//    DONT USE THIS
//    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
//        super.onCreate(savedInstanceState, persistentState)
//    }

    var count = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Log.e("TAG", "onCreate")
        btnAdd.setOnClickListener {
            count++
            tvCount.text = count.toString()
        }
    }

    override fun onStart() {
        Log.e("TAG", "onStart")
        super.onStart()
    }

    override fun onResume() {
        super.onResume()
        Log.e("TAG", "onResume")
    }

    override fun onSaveInstanceState(outState: Bundle?) {
        super.onSaveInstanceState(outState)

//        outState?.let {
//            it.putInt("COUNT",count)
//        }

        outState?.putInt("COUNT", count)
        Log.e("TAG", "onSaveInstanceState")
    }

    override fun onPause() {
        super.onPause()
        Log.e("TAG", "onPause")
    }

    override fun onStop() {
        super.onStop()
        Log.e("TAG", "onStop")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.e("TAG", "onDestroy")
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle?) {
        super.onRestoreInstanceState(savedInstanceState)
        Log.e("TAG", "onRestoreInstanceState")

        count = savedInstanceState?.getInt("COUNT") ?: 0
        tvCount.text = count.toString()
    }

}
